services/QuillLMS/app/services/demo/create_admin_report.rb

```diff
@@ -2,195 +2,78 @@
 
 class Demo::CreateAdminReport
 
-  def initialize(name, email_safe_school_name, teacher_email)
-    @name = name
-    @email_safe_school_name = email_safe_school_name
+  NUMBER_OF_CLASSROOMS_TO_DESTROY_SESSIONS_FOR = 20
+  RANGE_OF_NUMBER_OF_SESSIONS_TO_DESTROY = 14..28 # 10-20% of 140
+
+  def initialize(teacher_email, passed_data=nil)
     @teacher_email = teacher_email
+    @passed_data = passed_data
   end
 
   def call
     create_demo
   end
 
-  attr_reader :name, :email_safe_school_name, :teacher_email
-  private :name
-  private :email_safe_school_name
+  def data
+    @data ||= @passed_data || Demo::SessionData.new.admin_demo_data
+  end
+
+  attr_reader :teacher_email
   private :teacher_email
```

Comment:
Since :teacher_email is passed into the constructor, it's not clear what purpose this private access modifier is. 
 You could remove the private access modifier and then move attr_reader up to the top of the file.